src/v/datalake/record_multiplexer.cc

```diff
@@ -268,8 +288,22 @@ record_multiplexer::operator()(model::record_batch batch) {
     co_return ss::stop_iteration::no;
 }
 
+ss::future<writer_error> record_multiplexer::flush_writers() {
+    if (_error) {
+        co_return *_error;
+    }
+    auto result = co_await ss::coroutine::as_future(ss::max_concurrent_for_each(
+      _writers, 10, [](auto& entry) { return entry.second->flush(); }));
+    if (result.failed()) {
+        vlog(_log.warn, "Error flushing writers: {}", result.get_exception());
```

Comment:
@oleiman if the log level were set to ERROR then this log line wouldn't print. however, would `result.get_exception()` still be evaluated? if not, this would be an abandoned exceptional future error. presumably yes, but i dunno what all goes on in vlog and below with macros etc...

Reply:
yeah good call out. `vlog` macro expands to a direct call to 

```
    template <typename... Args>
    void warn(format_info_t<Args...> fmt, Args&&... args) noexcept {
        log(log_level::warn, std::move(fmt), std::forward<Args>(args)...);
    }
```

so the decision whether or not to log is on the other side of that function call, so the arguments should be evaluated right away. which is to say, the only thing that's lazy is the actual format string interpolation.

Reply:
got it, thanks! looks good